packages/protocol/contracts/governance/Validators.sol

```diff
@@ -308,14 +406,128 @@ contract Validators is IValidators, Ownable, ReentrancyGuard, Initializable, Usi
     return getLockedGold().getAccountTotalLockedGold(account) >= balanceRequirements.group;
   }
 
+  /**
+   * @notice Returns the parameters that goven how a validator's score is calculated.
+   * @return The parameters that goven how a validator's score is calculated.
+   */
+  function getValidatorScoreParameters() external view returns (uint256, uint256) {
+    return (validatorScoreParameters.exponent, validatorScoreParameters.adjustmentSpeed.unwrap());
+  }
+
+  /**
+   * @notice Returns the group membership history of a validator.
+   * @param account The validator whose membership history to return.
+   * @return The group membership history of a validator.
+   */
+  function getMembershipHistory(
+    address account
+  )
+    external
+    view
+    returns (uint256[] memory, address[] memory)
+  {
+    MembershipHistoryEntry[] memory entries = validators[account].membershipHistory.entries;
+    uint256[] memory epochs = new uint256[](entries.length);
+    address[] memory membershipGroups = new address[](entries.length);
+    for (uint256 i = 0; i < entries.length; i = i.add(1)) {
+      epochs[i] = entries[i].epochNumber;
+      membershipGroups[i] = entries[i].group;
+    }
+    return (epochs, membershipGroups);
+  }
+
+  /**
+   * @notice Updates a validator's score based on its uptime for the epoch.
+   * @param validator The address of the validator.
+   * @param uptime The Fixidity representation of the validator's uptime, between 0 and 1.
+   * @return True upon success.
+   */
+  function updateValidatorScore(address validator, uint256 uptime) external onlyVm() {
+    _updateValidatorScore(validator, uptime);
+  }
+
+  /**
+   * @notice Updates a validator's score based on its uptime for the epoch.
+   * @param validator The address of the validator.
+   * @param uptime The Fixidity representation of the validator's uptime, between 0 and 1.
+   * @return True upon success.
+   */
+  function _updateValidatorScore(address validator, uint256 uptime) internal {
+    address account = getLockedGold().getAccountFromValidator(validator);
+    require(isValidator(account), "isvalidator");
+    require(uptime <= FixidityLib.fixed1().unwrap(), "uptime");
+
+    uint256 numerator;
+    uint256 denominator;
+    (numerator, denominator) = fractionMulExp(
```

Comment:
it would be nice to document the math here

Reply:
also, it seems we are using just "half" of what this very specific precompile provides

Reply:
also.. now that we use fixidity the call seems suboptimal, the denomintaor is always "1"... and could be replaces by statint the number of decimal positions the numerator has

Reply:
this was discussed offline, and there's an issue to use "fixidity" on the precompile